mpsort-mpi.c

```diff
@@ -393,7 +217,9 @@ mpsort_mpi_newarray_impl (void * mybase, size_t mynmemb,
     size_t sizes[NTask];
     size_t outsizes[NTask];
     size_t myoffset;
-    size_t totalsize = _collect_sizes(mynmemb, sizes, &myoffset, comm);
+    size_t myoutoffset;
+    size_t totalsize = MPIU_Segmenter_collect_sizes(mynmemb, sizes, &myoffset, comm);
+    MPIU_Segmenter_collect_sizes(myoutnmemb, outsizes, &myoutoffset, comm);
```

Comment:
I hope this addresses issue#10.

Reply:
Should we really call the function twice? That will just set outsizes and sizes to the same thing. Maybe easier to just let outsizes be NULL in Segmenter_init, to indicate ignored?

Reply:
myoutnmemb and mynmemb are different, when we use MP-Sort to redistribute data evenly for example.

Reply:
Ah, I see. Perhaps a comment here then would be helpful